server/server.go

```diff
@@ -575,6 +575,10 @@ func NewServer(userConfig UserConfig, config Config) (*Server, error) {
 		Webhooks:                   webhooksManager,
 		WorkingDirLocker:           workingDirLocker,
 		AggregateApplyRequirements: applyRequirementHandler,
+		PullStatusFetcher:          boltdb,
+		StaleCommandChecker: &events.StaleCommandHandler{
```

Comment:
If we wanted to reduce upstream impact, one thing we could do is maintain addition of timestamps in the model, but provide a default StaleCommandChecker() that returns false, allowing people to bypass the check. Our implementation can instead exist within our own decorator package.

Reply:
I think what you have is fine, it shouldn't technically fail or matter for the default case.